interview-questions.md

```diff
@@ -8,36 +8,243 @@ Then, research the question to expand on your answer. Even if you feel you have
+You have successfully added a foreign key to students that pairs with corhort if you set up the associations. it is not part of the question asked so I will give a quick answer
+app/models/cohort.rb add has_many :students
+app/models/student.rb belongs_to :cohort
+
+    I had to do this like 30 min before I answered this question. that is why I did such an amazing job on the non-researched answer. 
+    Something that also needs to be kept in mind is the pluralizatoin of the model names.
+    The ability to use other association terms, and somewhat to do with this, the changing of route names when  having rails build them for you, which in turn goes back to pluralizatoin. 
+     This is a very simple rails project with an easy association and you forgot a foreign key. 
+     Now imagine if it was lots of associations and as you add new controllers names are changing and it can all get very confusing. 
+     for the routes aspect of it rails routes-E, is very helpful. 
+     and for this specific pluralization issue I keep running into and screwing up. 
+    I have to verbally say out loud, "a student doesnt have multiple cohorts, a cohort has multiple students." 
+    I say that a few times, then shake my head like im ready to start inputting this info, but the whole time im doing it and I get to a place that may be pluralized, I have to say it outloud again just to be sure. 
+     A small mistake on the this can be very annoying. 
+     Another major thing that I have run into. 
+     way worse than anything you just did by forgetting the foreiegn key, but kind of the same. 
+     I forgot the foreign key and also had rails generate all the routes for me. 
+     and then when I went back and added the foreign key to it as described above, I realized that the routes were not working correctly and that something was now screwed up. 
+     Due to the porjects being small it is a very possible option to just clean slate restart and go again from scratch, but that will not always be an option and I think it is important that I take the time to try and figure out how to solve these errors I am running into when I make a mistake and how to resolve them instead of starting over. 
+
+
+
 2. Which RESTful routes must always be passed params? Why?
+ Your answer: 
+    A param is attached to an instance variable and the param makes it so we can pass a value from a hash. So anything that needs a specific value will absolutely need a param, i am refering to :id.
+    Show, Update, Edit, and Destroy have to have params because they need to know what to show or what to edit/update, or what to delete, and that is figured out by using the params :id. I believe we do not "have" to have them when creating. i know that we use strong params but i dont think that we have to. see example below for a visual representation of what i stated
+            For example purposes, i am going to call the model Animal
+                The RESTful routes are:
```

Comment:
Look over your examples for the routes....still outstanding content.
https://github.com/learn-academy-2022-echo/Syllabus/blob/main/rails/restful-routes-crud.md